addons/website/models/res_config.py

```diff
@@ -1,51 +1,46 @@
-        'cdn_url': fields.related('website_id', 'cdn_url', type="char", string='CDN Base URL'),
-        'cdn_filters': fields.related('website_id', 'cdn_filters', type="text", string='CDN Filters'),
-        'module_website_form_editor': fields.boolean("Form builder: create and customize forms"),
-        'module_website_version': fields.boolean("A/B testing and versioning"),
-    }
-
-    def on_change_website_id(self, cr, uid, ids, website_id, context=None):
-        if not website_id:
-            return {'value': {}}
-        website_data = self.pool.get('website').read(cr, uid, [website_id], [], context=context)[0]
-        values = {'website_name': website_data['name']}
-        for fname, v in website_data.items():
-            if fname in self._columns:
-                values[fname] = v[0] if v and self._columns[fname]._type == 'many2one' else v
-        return {'value' : values}
-
-    # FIXME in trunk for god sake. Change the fields above to fields.char instead of fields.related, 
+    website_id = fields.Many2one('website', string="website", required=True, default=lambda self: self.env['website'].search([])[0].id)
+    website_name = fields.Char(related='website_id.name', string="Website Name")
+
+    language_ids = fields.Many2many(related='website_id.language_ids', relation='res.lang', string='Languages')
+    default_lang_id = fields.Many2one(related='website_id.default_lang_id', relation='res.lang', string='Default language')
+    default_lang_code = fields.Char(related='website_id.default_lang_code', string="Default language code")
+    google_analytics_key = fields.Char(related='website_id.google_analytics_key', string='Google Analytics Key')
+
+    social_twitter = fields.Char(related='website_id.social_twitter', string='Twitter Account')
+    social_facebook = fields.Char(related='website_id.social_facebook', string='Facebook Account')
+    social_github = fields.Char(related='website_id.social_github', string='GitHub Account')
+    social_linkedin = fields.Char(related='website_id.social_linkedin', string='LinkedIn Account')
+    social_youtube = fields.Char(related='website_id.social_youtube', string='Youtube Account')
+    social_googleplus = fields.Char(related='website_id.social_googleplus', string='Google+ Account')
+    compress_html = fields.Boolean(related='website_id.compress_html', string='Compress rendered HTML for a better Google PageSpeed result')
+    cdn_activated = fields.Boolean(related='website_id.cdn_activated', string='Use a Content Delivery Network (CDN)')
+    cdn_url = fields.Char(related='website_id.cdn_url', string='CDN Base URL')
+    cdn_filters = fields.Text(related='website_id.cdn_filters', string='CDN Filters')
+    module_website_form_editor = fields.Boolean(string="Website form builder")
+    module_website_version = fields.Boolean(string="Website A/B testing and versioning")
+
+    @api.onchange('website_id')
+    def on_change_website_id(self):
```

Comment:
Hello @hmo-odoo ,
it's used onchange of website_id(many2one with website model) need to set all fields of website_config_setting and there might be the field exist or may be many2one or simple so need to set all the fields value based on many2one of website_id so need this stuff.